tools/src/bin/build-blame.rs

```diff
@@ -659,78 +742,88 @@ fn main() {
             .collect::<Vec<_>>();
         let blame_parents = commit
             .parent_ids()
-            .map(|pid| blame_repo.find_commit(blame_map[&pid]).unwrap())
+            .map(|pid| blame_map[&pid])
             .collect::<Vec<_>>();
 
-        let mut builder = blame_repo.treebuilder(None).unwrap();
+        // Scope the import_helper borrow
+        {
+            let mut import_stream = BufWriter::new(import_helper.stdin.as_mut().unwrap());
+            write!(import_stream, "commit {}\n", blame_ref).unwrap();
+            write!(import_stream, "mark :{}\n", rev_done).unwrap();
+            blame_map.insert(*git_oid, BlameRepoCommit::Mark(rev_done));
+
+            let mut write_role = |role: &str, sig: &git2::Signature| {
+                write!(import_stream, "{} ", role).unwrap();
+                import_stream.write(sig.name_bytes()).unwrap();
+                write!(import_stream, " <").unwrap();
+                import_stream.write(sig.email_bytes()).unwrap();
+                write!(import_stream, "> ").unwrap();
+                let when = sig.when();
+                write!(
+                    import_stream,
+                    "{} {}{:02}{:02}\n",
+                    when.seconds(),
+                    when.sign(),
+                    when.offset_minutes().abs() / 60,
+                    when.offset_minutes().abs() % 60,
+                )
+                .unwrap();
+            };
+            write_role("author", &commit.author());
+            write_role("committer", &commit.committer());
+
+            let commit_msg = if let Some(hg_rev) = hg_rev {
+                format!("git {}\nhg {}\n", git_oid, hg_rev)
+            } else {
+                format!("git {}\n", git_oid)
+            };
+
+            write!(import_stream, "data {}\n{}\n", commit_msg.len(), commit_msg).unwrap();
+            if let Some(first_parent) = blame_parents.first() {
+                write!(import_stream, "from {}\n", first_parent).unwrap();
+            } else {
+                // This is a new root commit, so we need to use a special null
+                // parent commit identifier for git-fast-import to know that.
+                write!(
+                    import_stream,
+                    "from 0000000000000000000000000000000000000000\n"
+                )
+                .unwrap();
+            }
+            for additional_parent in blame_parents.iter().skip(1) {
+                write!(import_stream, "merge {}\n", additional_parent).unwrap();
+            }
+            write!(import_stream, "deleteall\n").unwrap();
```

Comment:
Potential comment: Start from a clean slate for simplicity. https://git-scm.com/docs/git-fast-import#_filedeleteall